src/xrpld/app/tx/detail/apply.cpp

```diff
@@ -113,12 +146,88 @@ apply(
     ApplyFlags flags,
     beast::Journal j)
 {
-    STAmountSO stAmountSO{view.rules().enabled(fixSTAmountCanonicalize)};
-    NumberSO stNumberSO{view.rules().enabled(fixUniversalNumber)};
+    return apply(app, view, [&]() mutable {
+        return preflight(app, view.rules(), tx, flags, j);
+    });
+}
 
-    auto pfresult = preflight(app, view.rules(), tx, flags, j);
-    auto pcresult = preclaim(pfresult, app, view);
-    return doApply(pcresult, app, view);
+std::pair<TER, bool>
+apply(
+    Application& app,
+    OpenView& view,
+    uint256 const& batchId,
+    STTx const& tx,
+    ApplyFlags flags,
+    beast::Journal j)
+{
+    return apply(app, view, [&]() mutable {
+        return preflight(app, view.rules(), batchId, tx, flags, j);
+    });
+}
+
+static bool
+applyBatchTransactions(
+    Application& app,
+    OpenView& batchView,
+    STTx const& txn,
+    beast::Journal j)
+{
+    XRPL_ASSERT(
+        txn.getTxnType() == ttBATCH &&
+            txn.getFieldArray(sfRawTransactions).size() != 0,
```

Comment:
```suggestion
            txn[~sfRawTransactions].value_or(0).size() != 0,
```
Otherwise I think `getFieldArray` will crash if `sfRawTransactions` doesn't exist as a field

Reply:
`sfRawTransactions` is a required field... but I can add this